### PR TITLE
SCAL-229204 defined types for dateFormatConfig in appConfig

### PR DESCRIPTION
**Pull request by UtkarshRai12**

### CODE REVIEW

File: src/types/common.types.ts [hunk 1]
```diff
@@ -250,9 +250,25 @@ export type ChartSdkCustomStylingConfig = {
...
+ * The date formats are used to format the date values in the chart
+ */
+
+export type ChartSdkDateFormatsConfig = {
+    tsLocaleBasedDateFormats?: Record<string, string>;
+    tsLocaleBasedStringsFormats?: Record<string, string>;
+    tsDateConstants?: Record<string, string>;
+    calanders?: any;
+    DEFAULT_DATASOURCE_ID?: any;
+};
```
Comment (harshmeetTS) [inline src/types/common.types.ts:267]: read about type and interface

File: src/types/common.types.ts [hunk 2]
```diff
@@ -250,9 +250,25 @@ export type ChartSdkCustomStylingConfig = {
...
+ * @remarks
+ * The date formats are used to format the date values in the chart
+ */
+
+export type ChartSdkDateFormatsConfig = {
+    tsLocaleBasedDateFormats?: Record<string, string>;
+    tsLocaleBasedStringsFormats?: Record<string, string>;
+    tsDateConstants?: Record<string, string>;
+    calanders?: any;
+    DEFAULT_DATASOURCE_ID?: any;
```
Comment (harshmeetTS) [inline src/types/common.types.ts:266]: everything should be camel case

Comment (UtkarshRai12) [reply src/types/common.types.ts]: ack

File: src/types/common.types.ts (hunk 1 above)
Comment (UtkarshRai12) [reply src/types/common.types.ts:296]: ack

**Review by harshmeetTS — APPROVED** [review]

File: example/custom-bar-chart/custom-chart.ts [hunk 3]
```diff
@@ -20,6 +21,7 @@ import {
     dateFormatter,
     getCfForColumn,
     getChartContext,
+    getCustomCalendarGuidFromColumn,
```
Comment (harshmeetTS) [inline example/custom-bar-chart/custom-chart.ts:24]: rt

File: src/types/common.types.ts [hunk 4]
```diff
@@ -250,9 +250,25 @@ export type ChartSdkCustomStylingConfig = {
...
+ * The date formats are used to format the date values in the chart
+ * @remarks
+ * The date formats are used to format the date values in the chart
+ */
+
+export type ChartSdkDateFormatsConfig = {
+    tsLocaleBasedDateFormats?: Record<string, string>;
+    tsLocaleBasedStringsFormats?: Record<string, string>;
+    tsDateConstants?: Record<string, string>;
+    calanders?: any;
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply src/types/common.types.ts]: ack

File: src/utils/date-formatting.ts [hunk 5]
```diff
@@ -81,29 +316,365 @@ export function getStartEpoch(date: CustomCalendarDate): number | null {
...
+    }
+    if (!_.isNumber(epochMillis)) {
+        console.log(
+            'formatDate could not convert input date to a timestamp',
+            inputDate,
+        );
+        return `${inputDate}`;
     }
-    return DateTime.fromMillis(dataValue * 1000).toFormat('dd-MM-yyyy');
+    console.log(formatPattern);
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply src/utils/date-formatting.ts:673]: ack

File: src/utils/formatting-util.spec.ts [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+        expect(DateFormatting.formatDate).toHaveBeenCalledWith(
+            1234567890,
+            dateFormatPresetsMock.DATETIME_SHORT_WITH_SECONDS,
+            true,
+            {
+                ...mockOptions,
+                customCalendarOverridesFiscalOffset: true,
+            },
+        );
+        // expect(formattedDateWithoutMillis).toBe('MM/DD/YYYY HH:mm:ss'); // Expect pattern without millis
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply src/utils/formatting-util.spec.ts:311]: ack

File: src/utils/formatting-util.ts [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    appConfig: any,
+    col: ChartColumn,
+    data: any,
+): any => {
+    let customCalenderMap = {};
+
+    if (
+        getCustomCalendarGuidFromColumn(col) !== null &&
+        getCustomCalendarGuidFromColumn(col) !== undefined &&
+        getCustomCalendarGuidFromColumn(col) !== ''
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply src/utils/formatting-util.ts:213]: ack

File: src/utils/formatting-util.ts [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+    ) {
+        for (let i = 0; i < data.length; i++) {
+            customCalenderMap = {
+                ...customCalenderMap,
+                [data[i].v.s]: data[i],
+            };
+        }
```
[the remark this answers is not in the document]
Comment (UtkarshRai12) [reply src/utils/formatting-util.ts:220]: ack

**Pull request merged**